Public/Src/Cache/ContentStore/Library/Service/Grpc/GrpcCopyClient.cs

```diff
@@ -339,9 +281,9 @@ private async Task<T> RunClientActionAndThrowIfFailedAsync<T>(Context context, F
         /// <inheritdoc />
         public void Dispose()
         {
-            lock (this)
+            if (!ShutdownStarted || !ShutdownCompleted)
             {
-                _uses--;
+                throw new CacheException($"{nameof(GrpcCopyClient)} must be shutdown before disposing.");

```

Comment:
I'm not sure, but I think it should be fine to dispose the insance if ShutdownAsync fails.

But I think this code will fail.

Should it be:

`if (ShutdownStarted && !ShutdownCompleted) `